app/controllers/BaseController.scala

```diff
@@ -36,18 +36,24 @@ trait BaseController extends FrontendBaseController with I18nSupport with Enumer
                  (implicit request: DataRequest[_], format: Format[A]): Form[A] =
     request.userAnswers.get(page).fold(form)(form.fill)
 
-  def withItem(idx: Int)(f: MovementItem => Result)(implicit request: DataRequest[_]): Result =
+  def withAllCompletedItemsAsync()(f: Seq[MovementItem] => Future[Result])(implicit request: DataRequest[_]): Future[Result] =
+    f(request.getAllCompletedItemDetails)
+
+  def withAddedItem(idx: Int)(f: MovementItem => Result)(implicit request: DataRequest[_]): Result =
     request.getItemDetails(idx) match {
       case Some(item) => f(item)
       case None => Redirect(routes.SelectItemsController.onPageLoad(request.ern, request.arc).url)
     }
```

Comment:
Tbh this is only used in one place - can probably be removed in favour of `withAddedItemAsync` (and a `Future.successful(...)` wherever it's used)